src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ArrayOfEmbeddedDataNode.cs

```diff
@@ -23,12 +23,14 @@ public class ArrayOfEmbeddedDataNode<TEmbedded> : ObjectNode
         private ObjectAndOffsetSymbolNode _startSymbol;
         private ObjectAndOffsetSymbolNode _endSymbol;
         private IComparer<TEmbedded> _sorter;
+        private EmbeddedObjectNode _endOfArraySentinel;
```

Comment:
I don't particularly like the extensibility points this forces us to do. Maybe just having a special node kind wouldn't be that bad. ArrayOfEmbeddedObjects will have it's own big set of problems when we go multithreaded already...